File: backend/models/place.js
const mongoose = require("mongoose");


const placeSchema = new mongoose.Schema({


    name:{
        type:String,
        required:true
    },


    shortDescription:String,


    description:String,


    city:String,


    state:String,


    country:String,


    location:String,


    imageUrl:String,



    addedBy:{
        type:mongoose.Schema.Types.ObjectId,
        ref:"User",
        required:true
    },



    ratings:[

        {

            user:{
                type:mongoose.Schema.Types.ObjectId,
                ref:"User"
            },


            value:{
                type:Number,
                min:1,
                max:5
            }

        }

    ],



    averageRating:{
        type:Number,
        default:0
    },



    visitorsCount:{
        type:Number,
        default:0
    },



    // Admin Approval System

    approvalStatus:{

        type:String,

        enum:[
            "pending",
            "approved",
            "rejected"
        ],

        default:"pending"

    },



    adminFeedback:{

        type:String,

        default:""

    },



    reviewedBy:{

        type:mongoose.Schema.Types.ObjectId,

        ref:"User"

    },



    reviewedAt:{

        type:Date

    },



    createdAt:{

        type:Date,

        default:Date.now

    }


});


// Collection name "places" (Mongoose default plural of "Place").
// Older seeded documents live here. Forced "place" returned empty lists.
module.exports = mongoose.model("Place", placeSchema, "places");

File: android/app/src/main/java/com/example/travira/model/Place.kt
package com.example.travira.model


/**
 * Matches the live API response from GET /api/places
 * (raw JSON array with rating field).
 */
data class Place(
    val _id: String = "",
    val id: Int = 0,
    val name: String = "",
    val shortDescription: String? = null,
    val description: String? = null,
    val city: String? = null,
    val state: String? = null,
    val country: String? = null,
    val location: String? = null,
    val imageUrl: String? = null,
    val rating: Double = 0.0,
    val averageRating: Double = 0.0,
    val visitorsCount: Int = 0,
    val approvalStatus: String? = null
) {
    /** Prefer live `rating`, fall back to averageRating */
    val displayRating: Double
        get() = if (rating > 0) rating else averageRating
}
